### PR TITLE
Add dwave-greedy docs

### PR DESCRIPTION
**Pull request by JoelPasvolsky**

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -34,3 +34,6 @@
 [submodule "dwave-inspector"]
 	path = dwave-inspector
 	url = https://github.com/dwavesystems/dwave-inspector.git
+[submodule "dwave-greedy"]
+	path = dwave-greedy
+	url = git@github.com:dwavesystems/dwave-greedy.git
```
Comment (randomir) [inline .gitmodules:39]: Please use the https url (see above).

**Review by randomir — CHANGES_REQUESTED** [review]
We'll also want to add it to `setup.py`, but need to wait with that until we release it.

File: .gitmodules (hunk 1 above)
Comment (JoelPasvolsky) [reply .gitmodules]: Thanks, fixed

**Comment by JoelPasvolsky** [comment]
doctest failure should be fixed by https://github.com/dwavesystems/dwave-system/pull/342

File: docs/examples/pp_greedy.rst [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+This example uses explicit postprocessing to improve results returned 
+from a quantum computer. 
+
+Typically, Ocean tools do some minimal, implicit postprocessing; for 
+example, when you use 
+`embedding tools <https://docs.ocean.dwavesys.com/en/stable/docs_system/reference/embedding.html>`_
+to map problem variables to qubits, *broken* :term:`chain`\ s (differing
+states of the qubits representing a variable) may be resolved by majority 
+vote: Ocean sets the variable's value based on the state returned from
```
Comment (arcondello) [inline docs/examples/pp_greedy.rst:15]: seems like a good opportunity to link to majority vote function

**Review by arcondello — APPROVED** [review]

File: docs/examples/pp_greedy.rst (hunk 2 above)
Comment (JoelPasvolsky) [reply docs/examples/pp_greedy.rst:15]: I linked to the page a bit earlier in the example: 
```
when you use 
`embedding tools <https://docs.ocean.dwavesys.com/en/stable/docs_system/reference/embedding.html>`_
to map problem variables to qubits, *broken* :term:`chain`\ s (differing states of the qubits representing a variable) may be resolved by majority vote
```
I'm chary of linking to anything lower level that might move and break link

File: docs/examples/pp_greedy.rst [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+.. testcode::
+
+    from greedy import SteepestDescentSolver
+
+    solver_greedy = SteepestDescentSolver()
+
+    sampleset_qpu = sampler.sample_ising(h, J, num_reads=100, answer_mode='raw')
+    # Postprocess
+    sampleset_pp = solver_greedy.sample_ising(h, J, initial_states=sampleset_qpu)
```
Comment (randomir) [inline docs/examples/pp_greedy.rst:85]: How about renaming `sampler` to `qpu` and then:
```suggestion
    import greedy

    postprocessor = greedy.SteepestDescentSolver()

    sampleset_qpu = qpu.sample_ising(h, J, num_reads=100, answer_mode='raw')
    sampleset_pp = postprocessor.sample_ising(h, J, initial_states=sampleset_qpu)
```

**Review by randomir — APPROVED** [review]
:+1:  for adding the post-processing example.

Please update submodule to the latest commit/6d47a112d. I like doing it like this:
```
cd dwave-greedy
git fetch --all
git checkout 0.1.0
```

**Comment by JoelPasvolsky** [comment]
Doctest finally pass--PR is ready to merge.

**Pull request merged**